### ISSUE TITLE
assess flutter_gallery `use_build_context_synchronously` breakages 

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#3680]
A recent fix to `use_build_context_synchronously` (#3679) is flagging a few places in `flutter_gallery` that need addressing before we can publish a new linter.

https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8803460924190469121/+/u/analyze_flutter_flutter/stdout

Naively I added some `mounted` checks but these won't work since the widgets are _stateless_ and don't have a mounted property accessor (https://github.com/flutter/flutter/pull/111321). We could make these diagnostics go away by limiting the lint to only check accesses from `StatefulWidget`s (indeed the [lint docs](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html) suggest this) but there's also something fishy going on in the newly flagged calls...  I'd love another set of eyes and 2 cents.

Thanks in advance!

/cc @a14n @goderbauer @jacob314 

/fyi @raapperez

**Comment by pq** [comment]
(Marking `P1` as this is currently roll-blocking.)

**Comment by goderbauer** [comment]
The problem also exists for contexts of StatelessWidgets, though. One workaround is to turn the StatelessWidget into a StatefulWidget so you get access to the `mounted` property. I've also been thinking about adding a property to `BuildContext` itself so you can check right on it whether it is still valid; discussion for that is at https://github.com/flutter/flutter/issues/111488.

**Comment by a14n** [comment]
https://github.com/flutter/flutter/pull/111539 fixes the issue.

**Comment by pq** [comment]
Thanks @a14n!

**Comment by pq** [comment]
Looks like we've still got a failure:

https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8803023674715529281/+/u/analyze_flutter_gallery/stdout

```
Analyzing gallery...

   info - lib/pages/demo.dart:198:13 - Do not use BuildContexts across async gaps. - use_build_context_synchronously
```

**Comment by a14n** [comment]
https://github.com/flutter/gallery/pull/782 fixes this new issue.

**Comment by pq** [comment]
Thanks for the quick fix!

**Comment by pq** [comment]
<img width="558" alt="image" src="https://user-images.githubusercontent.com/67586/190703993-86522062-85d5-4520-882f-a8f6141dcf14.png">

Thanks!
